src/main/java/org/spongepowered/api/command/AbstractCommandModule.java

```diff
@@ -0,0 +1,365 @@
+        this.bind(Game.class).toProvider((args, modifiers) -> Sponge.getGame());
+    }
+
+    private void noChangingRegistration(String what) {
+        checkState(!this.registered, "The %s cannot be changed after commands have been registered", what);
+    }
+
+    /**
+     * Determines if this command module can be installed or configured
+     * through a {@link CommandManager}.
+     *
+     * @param install {@code true} if we should check if this module
+     *     can be installed
+     * @return {@code true} if this command module can be installed
+     *     or configured
+     */
+    @SuppressWarnings("ConstantConditions")
+    public final boolean canBeInstalledOrConfigured(boolean install) {
+        return install ? (this.plugin == null && this.manager == null && !this.configured) : !this.configured;
+    }
+
+    /**
+     * Configure this command module.
+     *
+     * <p>You <b>must</b> {@link #register(Object) register commands} <i>after</i>
+     * you have configured any bindings, input tokenizer, etc.</p>
+     */
+    @SuppressWarnings("WeakerAccess")
+    protected abstract void configure();
+
+    /**
+     * Create a binding builder from a class type.
+     *
+     * @param clazz The class
+     * @param <T> The bound type
+     * @return A binding builder
+     */
+    protected final <T> BindingBuilder<T> bind(Class<T> clazz) {
+        this.noChangingRegistration("bindings");
+        return this.binder.bind(clazz);
+    }
+
+    /**
+     * Create a binding builder from a binding key.
+     *
+     * @param key The binding key
+     * @param <T> The bound type
+     * @return A binding builder
+     */
+    protected final <T> BindingBuilder<T> bind(BindingKey<T> key) {
+        this.noChangingRegistration("bindings");
+        return this.binder.bind(key);
+    }
+
+    /**
+     * Registers all found {@link Command annotation commands}.
+     *
+     * <p>You <b>must</b> register commands <i>after</i> you have
+     * configured any bindings, input tokenizer, etc.</p>
+     *
+     * @param object The object to register commands from
+     */
+    protected final void register(Object object) {
+        this.registered = true;
+        this.manager.register(this, object);
+    }
+
+    /**
+     * Register a given command using the given list of aliases.
+     *
+     * <p>You <b>must</b> register commands <i>after</i> you have
+     * configured any bindings, input tokenizer, etc.</p>
+     *
+     * <p>If there is a conflict with one of the aliases (i.e. that alias
+     * is already assigned to another command), then the alias will be skipped.
+     * It is possible for there to be no alias to be available out of
+     * the provided list of aliases, which would mean that the command would not
+     * be assigned to any aliases.</p>
+     *
+     * <p>The first non-conflicted alias becomes the "primary alias."</p>
```

Comment:
I've wondered the same thing about the register method in CommandManager as well.